### PR TITLE
Add support for image capture

### PR DESCRIPTION
**Pull request by vishnuravi**
# Add support for image capture

## :recycle: Current situation & Problem
Multiple projects using CardinalKit require a method for allowing image capture within FHIR questionnaires rendered by ResearchKitOnFHIR.

## :bulb: Proposed solution
The FHIR SDC implementation guide defines an 'attachment' questionnaire item type, which is meant to be rendered as a binary file upload, including images, PDFs or other file types. ResearchKit does not support arbitrary binary file upload, but we can map this type to an `ORKImageCaptureStep` in order to allow users to capture images.

### Code of Conduct & Contributing Guidelines 

By submitting creating this pull request, you agree to follow our [Code of Conduct](https://github.com/StanfordBDHG/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordBDHG/.github/blob/main/CONTRIBUTING.md):
- [X] I agree to follow the [Code of Conduct](https://github.com/StanfordBDHG/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordBDHG/.github/blob/main/CONTRIBUTING.md).

### CODE REVIEW

**Comment by vishnuravi** [comment]
Note that it is not possible to create a UI test for this PR because `ORKImageCaptureStep` requires access to the device camera which cannot be simulated.

**Comment by vishnuravi** [comment]
Example output:
IMG_0568

File: Example/ExampleApp/Questionnaire/ORKOrderedTaskView.swift [hunk 1]
```diff
@@ -24,13 +25,44 @@ struct ORKOrderedTaskView: UIViewControllerRepresentable {
     init(
```
Comment (PSchmiedmayer) [inline Example/ExampleApp/Questionnaire/ORKOrderedTaskView.swift:25]: If would be great if we can add documentation for the `tintColor` and `outputDirectory` parameters to fully document the initializer.

File: Example/ExampleApp/Questionnaire/ORKOrderedTaskView.swift [hunk 2]
```diff
@@ -24,13 +25,44 @@ struct ORKOrderedTaskView: UIViewControllerRepresentable {
...
+        self.outputDirectory = outputDirectory
+    }
+
+    func getTaskOutputDirectory(_ taskViewController: ORKTaskViewController) -> URL? {
+        do {
+            let defaultFileManager = FileManager.default
+
+            // Identify the documents directory.
+            let documentsDirectory = try defaultFileManager.url(
+                for: .documentDirectory,
```
Comment (PSchmiedmayer) [inline Example/ExampleApp/Questionnaire/ORKOrderedTaskView.swift:43]: I would suggest that we use the `.cachesDirectory` for storing the images. The documents directly might be synced or actively used by an application that uses the framework to store user information which might conflict with your use case.

Theoretically, the `.itemReplacementDirectory` parameter would be the best choice here as we should just hold on to the files for a very short time but using the `.cachesDirectory` would also be a reasonable choice with a bit longer guarantee that files would be persisted a bit longer. Depends on how we decide on my comment below.

We should maybe add an extension method to `ORKTaskViewController` to delete the files in the temporary directory. We can add to the documentation that this should be called within the `ORKTaskViewControllerDelegate` subclass's `taskViewController(_ taskViewController: ORKTaskViewController, didFinishWith reason: ORKTaskViewControllerFinishReason, error: Error?)` method after extracting the result and make it the responsibility of the caller to move the files to a directory where they can be persisted for a longer time. We don't want to be responsible for sensitive user data lying somewhere on disk as part of this framework and should put that responsibility to the user of the framework and therefore clean out the directory after we have passed the files on to the caller. If that would be the approach, I would suggest using `.itemReplacementDirectory`.

What do you think about this approach?

File: Example/Example-Info.plist [hunk 3]
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (PSchmiedmayer) [inline Example/Example-Info.plist:1]: I think that the Example-Info.plist can be removed as it does not contain any additional information. You will have to remove it from the Xcode Project configuration and this the builds and archives fine without any problems on my machine.

File: Example/Example-Info.plist.license [hunk 4]
```diff
@@ -0,0 +1,5 @@
+This source file is part of the ResearchKitOnFHIR open source project
```
Comment (PSchmiedmayer) [inline Example/Example-Info.plist.license:1]: Remove this file once the `.plist` file is removed.

File: Example/Example.xcodeproj/project.pbxproj [hunk 5]
```diff
@@ -422,7 +422,9 @@
 				DEVELOPMENT_TEAM = "";
 				ENABLE_PREVIEWS = YES;
 				GENERATE_INFOPLIST_FILE = YES;
+				INFOPLIST_FILE = "Example-Info.plist";
```
Comment (PSchmiedmayer) [inline Example/Example.xcodeproj/project.pbxproj:425]: Can be removed once you remove the `.plist` file

File: Example/Example.xcodeproj/project.pbxproj [hunk 6]
```diff
@@ -458,7 +460,9 @@
...
```
Comment (PSchmiedmayer) [inline Example/Example.xcodeproj/project.pbxproj:463]: Can be removed once you remove the `.plist` file

**Review by PSchmiedmayer — CHANGES_REQUESTED** [review]
Thank you for the addition @vishnuravi! Great to see this in the framework. I have one bigger comment regarding the temporary file location + storage mechanism behind this that I would look to get your opinion and input on.

Once we have identified a good solution there I think that there is nothing standing in the way of merging the PR. A very good code quality and structure; nothing to add there; great work! 🚀

File: Example/ExampleApp/Questionnaire/ORKOrderedTaskView.swift (hunk 1 above)
Comment (vishnuravi) [reply Example/ExampleApp/Questionnaire/ORKOrderedTaskView.swift:27]: Good catch! Addressed by [0adf95f](https://github.com/StanfordBDHG/ResearchKitOnFHIR/pull/59/commits/0adf95ff479a250443d7d294353aa94839993a9c)

File: Example/Example-Info.plist (hunk 3 above)
Comment (vishnuravi) [reply Example/Example-Info.plist]: Removed in [c7d76ef](https://github.com/StanfordBDHG/ResearchKitOnFHIR/pull/59/commits/c7d76efcf6d5367a838f0df9a6559f719d71bf9a)

File: Example/Example.xcodeproj/project.pbxproj (hunk 5 above)
Comment (vishnuravi) [reply Example/Example.xcodeproj/project.pbxproj]: Removed in [c7d76ef](https://github.com/StanfordBDHG/ResearchKitOnFHIR/pull/59/commits/c7d76efcf6d5367a838f0df9a6559f719d71bf9a)

File: Example/Example.xcodeproj/project.pbxproj (hunk 6 above)
Comment (vishnuravi) [reply Example/Example.xcodeproj/project.pbxproj]: Removed in [c7d76ef](https://github.com/StanfordBDHG/ResearchKitOnFHIR/pull/59/commits/c7d76efcf6d5367a838f0df9a6559f719d71bf9a)

File: Example/Example-Info.plist.license (hunk 4 above)
Comment (vishnuravi) [reply Example/Example-Info.plist.license]: Removed in [c7d76ef](https://github.com/StanfordBDHG/ResearchKitOnFHIR/pull/59/commits/c7d76efcf6d5367a838f0df9a6559f719d71bf9a)

File: Example/ExampleApp/Questionnaire/ORKOrderedTaskView.swift (hunk 2 above)
Comment (vishnuravi) [reply Example/ExampleApp/Questionnaire/ORKOrderedTaskView.swift:45]: This sounds like a reasonable approach to me. In the case that we use a temporary directory, the caller would not only be responsible for moving the files to another directory for long-term storage, but also to update the `QuestionnaireResponse` object with the new file URL.

Comment (PSchmiedmayer) [reply Example/ExampleApp/Questionnaire/ORKOrderedTaskView.swift:45]: Exactly, that is an other very good point! 🚀👍

**Review by PSchmiedmayer — APPROVED** [review]
A great job @vishnuravi! Thank you for incorporating all the feedback in my last review!
I like the new structure and mechanism to store these files in a temporary location 🚀

**Pull request merged**